osu.Game/Rulesets/Mods/ModSnapColour.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Localisation;
+
+namespace osu.Game.Rulesets.Mods
+{
+    /// <summary>
+    /// Mod that colours hitobjects based on the musical division they are on
+    /// </summary>
+    public class ModSnapColour : Mod
+    {
+        public override string Name => "Snap Colour";
+        public override string Acronym => "SC";
```

Comment:
Not sure on the naming of this thing. The existing mania setting is named "timing-based colouring". I would like to keep that naming consistent, but "timing-based colouring" a bit too much of a mouthful to use for a mod name. "Snap" is a bit of an editor term and I'm not sure it should be used in the mod name.

Maybe something like "Coloured Rhythm" could work? Or - going off the wall a bit - ["Synesthesia"](https://en.wikipedia.org/wiki/Synesthesia)?

Reply:
Yeah, the name was just taken from the function name - I figured that was some "standard" name for it in VSRGs but I'm not tied to it at all. I really like the suggestion of "Synesthesia", a cool reference and in-line with the other mod names. I've renamed it to Synesthesia in the latest commit.